File: TESTING/schema/flashcard.schema.ts
import mongoose from "mongoose";
import { SRS, SRSSchema, type ISRS } from "./SRS";
const { Schema } = mongoose;

export interface IFlashCard {
  front: any;
  back: any;
  SRS: ISRS;
  inCollection: String;
  // _id?: string;
}

export const flashCardSchema = new Schema<IFlashCard>(
  {
    front: { type: Schema.Types.Mixed },

    back: { type: Schema.Types.Mixed },

    inCollection: Schema.Types.ObjectId,

    SRS: { type: SRSSchema, default: () => ({}) },
  },
  { timestamps: true }
);

export const FlashCard = mongoose.model("Flashcard", flashCardSchema);

File: test/test-database-go/pkg/mongodb/schema/FlashCard.schema.go
package schema

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

// SRS is a placeholder for your SRS schema
// Define the SRS struct according to its fields
type SRS struct {
    // Define your SRS fields here
}

// FlashCard represents the MongoDB flash card document
type FlashCard struct {
    ID          primitive.ObjectID `bson:"_id,omitempty"` // MongoDB ObjectID
    Front       interface{}        `bson:"front"`         // Mixed type, use interface{} for any type
    Back        interface{}        `bson:"back"`          // Mixed type
    SRS         *SRS              `bson:"SRS,omitempty"` // Optional reference to SRS
    InCollection primitive.ObjectID `bson:"inCollection"`  // Reference to Collection
}